core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -565,8 +565,9 @@ private[spark] class BlockManager(
             // Give up trying anymore locations. Either we've tried all of the original locations,
             // or we've refreshed the list of locations from the master, and have still
             // hit failures after trying locations from the refreshed list.
-            throw new BlockFetchException(s"Failed to fetch block after" +
-              s" ${totalFailureCount} fetch failures. Most recent failure cause:", e)
+            logError(s"Failed to fetch block after $totalFailureCount fetch failures." +
```

Comment:
This log statement corresponds to the case where we've tried to fetch the block from all known locations (or `maxFetchFailures` of them) and all of those fetches have failed with errors. Failures to fetch from only _one_ location are logged at warning level a few lines down from here.

This is probably worth downgrading to a warning to prevent users from becoming confused: I've noticed that people sometimes report the first error that they find in a log as the cause of a job failure even when the real failure occurs elsewhere. Warning level will also help filter this out in automated log analysis to search for errors.